### DISCUSSION TITLE
Using the library in rust

### DISCUSSION THREAD

**Discussion by elvinasn** [trustwallet/wallet-core#3905] [Q&A]
Hello, how does one use this library in rust? I saw the sample code, but in the example it wrote all of the C headers, instead of adding the rust folder as a crate. Or is that not possible?

**Comment by satoshiotomakan** [comment]
Hi @elvinasn, thank you for the question.
We do have plans to publish our crates to crates.io, but you can still use our Rust crates directly.
You only need to clone the whole https://github.com/trustwallet/wallet-core/ repository (no matter where to clone it).
Then you can import all required crates in Cargo.toml like this:
```toml
[dependencies]
tw_evm = { path = "path/to/wallet-core/rust/tw_evm" }
```

And then:
```rust
// main.rs
use tw_evm::address::Address;
use std::str::FromStr;

fn main() {
  let _ = Address::from_str("0xff00000000000000000000000000000000002694");
}
```

**Reply by saltyskip** [reply]
Any updates on getting this on crates.io

**Comment by Tolsi** [comment]
Do you happen to have any updates about crates.io? 💤

**Reply by satoshiotomakan** [reply]
Hi @Tolsi, we consider prioritising publishing the crates to crates.io this summer, thank you for your patience
